File: frontend/boutique_app/src/models/room.ts
import Beds from "./beds";
import Extras from "./extras";

export default class Room {
  id: number;
  name: string;
  // pricePerNightString: string;
  pricePerNight: number;
  type: string;
  description: string;
  beds: Beds;
  extras: Extras[];
  imageUrl: string;
  available: boolean | null;

  constructor(
    id: number,
    name: string,
    description: string,
    pricePerNight: number,
    type: string,
    beds: Beds,
    extras: Extras[],
    imageUrl: string,
    available: boolean | null
  ) {
    (this.id = id),
      (this.name = name),
      (this.pricePerNight = pricePerNight),
      (this.type = type),
      (this.description = description),
      (this.beds = beds),
      (this.extras = extras),
      (this.imageUrl = imageUrl),
      (this.available = available);
  }

  getLocalImage(): string {
    return new URL(`../assets/${this.imageUrl}`, import.meta.url).pathname;
  }
}

File: backend/boutique_app/src/main/kotlin/at/fhtw/dtos/responses/Room.kt
package at.fhtw.dtos.responses

data class Room(
    val id: Long,
    val name: String,
    val pricePerNight: Double,
    val type: String,
    val description: String,
    val beds: List<Bed>,
    val extras: Set<Extra>,
    val imageUrl: String?,
    val available: Boolean? = null,
) {
    companion object {
        fun from(old: at.fhtw.model.entities.Room, available: Boolean? = null): Room {
            return Room(
                old.id,
                old.name,
                old.pricePerNight,
                old.type.name,
                old.type.description,
                old.beds.map { Bed.from(it.bedType, it.amount) },
                old.extras.plus(old.type.standardExtras).map { Extra.from(it) }.toSet(),
                old.imageUrL,
                available
            )
        }
    }
}


